python_modules/dagster-graphql/dagster_graphql/implementation/asset_checks_loader.py

```diff
@@ -169,128 +172,3 @@ def get_checks_for_asset(
-    instance: DagsterInstance,
-    asset_record: Optional[AssetRecord],
-    execution: AssetCheckExecutionRecord,
-    resolved_status: AssetCheckExecutionResolvedStatus,
-) -> bool:
-    # always show in progress checks
-    if resolved_status == AssetCheckExecutionResolvedStatus.IN_PROGRESS:
-        return True
-
-    latest_materialization = (
-        asset_record.asset_entry.last_materialization_record if asset_record else None
-    )
-
-    if not latest_materialization:
-        # asset hasn't been materialized yet, so no reason to hide the check
-        return True
-
-    # If the check is executed in the same run as the materialization, then show it.
-    # This is a workaround to support the 'stage then promote' graph asset pattern,
-    # where checks happen before a materialization.
-    latest_materialization_run_id = latest_materialization.event_log_entry.run_id
-    if latest_materialization_run_id == execution.run_id:
-        return True
-
-    if resolved_status in [
-        AssetCheckExecutionResolvedStatus.SUCCEEDED,
-        AssetCheckExecutionResolvedStatus.FAILED,
-    ]:
-        evaluation = cast(
-            AssetCheckEvaluation,
-            check.not_none(check.not_none(execution.event).dagster_event).event_specific_data,
-        )
-        if not evaluation.target_materialization_data:
-            # check ran before the materialization was created
-            return False
-
-        # if the check matches the latest materialization, then show it
-        return (
-            evaluation.target_materialization_data.storage_id == latest_materialization.storage_id
-        )
-
-    # in this case the evaluation didn't complete, so we don't have target_materialization_data
-    elif resolved_status in [
-        AssetCheckExecutionResolvedStatus.EXECUTION_FAILED,
-        AssetCheckExecutionResolvedStatus.SKIPPED,
-    ]:
-        # As a last ditch effort, check if the check's run was launched after the materialization's
-        latest_materialization_run_record = instance.get_run_record_by_id(
-            latest_materialization_run_id
-        )
-        execution_run_record = instance.get_run_record_by_id(execution.run_id)
-        return bool(
-            latest_materialization_run_record
-            and execution_run_record
-            and execution_run_record.create_timestamp
-            > latest_materialization_run_record.create_timestamp
-        )
-
-    else:
-        check.failed(f"Unexpected check status {resolved_status}")
-
-
-class AssetChecksExecutionForLatestMaterializationLoader:
-    def __init__(self, instance: DagsterInstance, check_keys: List[AssetCheckKey]):
-        self._instance = instance
-        self._check_keys = check_keys
-        self._executions: Optional[
-            Mapping[AssetCheckKey, Optional[GrapheneAssetCheckExecution]]
-        ] = None
-
-    def _fetch_executions(self) -> Mapping[AssetCheckKey, Optional[GrapheneAssetCheckExecution]]:
```

Comment:
This method (and the entire class) is no longer necessary as the caching behavior is handled by the respective DataLoaders.

This is actually a great case study for why the DataLoader pattern (even if it's synchronous) is nice -- in this version of the code, we were doing a completely separate `get_asset_records` call and caching the results, when in reality we likely already fetched those asset records for some other purpose.

When using DataLoaders, we get to cache asset records (and whatever else) in a very lightweight way -- no need to thread around different BatchAssetRecordLoaders.